doc/srv6/srv6_path_tracing_midpoint.md

```diff
@@ -0,0 +1,438 @@
+
+		
+## Testing Requirements/Design  
+
+### Unit Test cases  
+
+#### Test cases for configuration CLI commands
+
+A new unit test `config_int_pt_test.py` is created to verify the new `config interface pt ...` CLI command introduced in SONiC.
+
+This test includes two test cases:
+
+- `test_interface_pt_interface_id_check` provides a valid PT Midpoint configuration to SONiC via the CLI and verifies that the SONiC CLI processes the command correctly (i.e., no errors are returned).
+- `test_interface_invalid_pt_interface_id_check` provides an invalid Interface ID and Timestamp Template to SONiC and verifies that the SONiC CLI rejects the configuration.
+
+#### Test cases for configuration show commands
+
+The existing `intfutil_test.py` is extended to verify the new `show interfaces pt [interface_name]` CLI command introduced in SONiC.
+
+Two new test cases are added:
+
+- `test_show_interfaces_pt_status` verifies that the `show interfaces pt` returns the expected output.
+- `test_show_interfaces_pt_Ethernet9_status` verifies that the `show interfaces pt Ethernet9` returns the expected output.
+
+#### Test cases for PortMgr
+
+To validate PT Midpoint, two test cases are added to the existing PortMgr unit test (`portmgr_ut.cpp`):
+
+- `ConfigurePortPTDefaultTimestampTemplate` writes a port configuration to the CONFIG_DB (without specifying the PT Timestamp Template) and verifies that PortMgr correctly propagates the PT Midpoint parameters to the APP_DB.
+- `ConfigurePortPTTimestampTemplate2` writes a port configuration to the CONFIG_DB (with a specific PT Timestamp Template) and verifies that PortMgr correctly propagates the PT Midpoint parameters to the APP_DB.
+
+#### Test cases for OrchAgent
+
+To validate PT Midpoint, two test cases are added to the existing PortsOrch unit test (`portsorch_ut.cpp`):
+
+- `PortPTConfigDefaultTimestampTemplate` writes a port configuration to the APPL_DB (with PT Interface ID non-zero and default PT Timestamp Template) and verifies that PortsOrch propagates the PT Midpoint parameters to the ASIC_DB.
+- `PortPTConfigTimestampTemplate2` writes a port configuration to the APPL_DB (with PT Interface ID non-zero and PT Timestamp Template specified) and verifies that PortsOrch correctly propagates the PT Midpoint parameters to the ASIC_DB.
+
+### System Test cases
+
+To validate PT Midpoint, two test cases are added to the existing `tests/test_port.py` (in the `sonic-swss` repository):
+
+- `test_PortPtDefaultTimestampTemplate` writes a port configuration to the CONFIG_DB (with PT Interface ID non-zero and default PT Timestamp Template) and verifies that APPL_DB and ASIC_DB are updated correctly.
+- `test_PortPtTimestampTemplate2` writes a port configuration to the CONFIG_DB (with PT Interface ID non-zero and PT Timestamp Template specified) and verifies that APPL_DB and ASIC_DB are updated correctly.
+
+## Open/Action items
+
+The PT Midpoint feature proposed in this document depends on the following PRs:
```

Comment:
Can SRv6 midpoint pt feature records switchId addition to interface ?

Reply:
@madhupalu, the Path Tracing midpoint records what is called Midpoint Compressed Data (MCD) [1]. The MCD contains Interface ID, Truncated Timestamp, and Interface Load. The MCD is recorded in the Path Tracing Hop-by-Hop header [2] which is optimzed for this MCD format. The switchId isn't recorded. 

[1] https://datatracker.ietf.org/doc/html/draft-filsfils-spring-path-tracing-04#name-midpoint-compressed-data
[2] https://datatracker.ietf.org/doc/html/draft-filsfils-spring-path-tracing-04#name-ipv6-hop-by-hop-path-tracin